### ISSUE TITLE
Enhancer makes typing laggy and is affected by windows size

### ISSUE THREAD

**Issue by kursorbyte** [notion-enhancer/notion-enhancer#344]
**describe the bug**
When the enhancer is applied, typing becomes slow and laggy. There is a lag between quickly pressing keys and the characters appearing on screen. It seems to be affected by the enhancer itself as the problem exhibits itself even when all mods are disabled (including core mods). 
Oddly enough, the lag worsens depending on Notion's window size. When full-screened, the lag is almost negligible (comparable to the un-enhanced app). When the window is resized to half the screen or smaller, the lag intensifies substantially. When the lag occurs and a string of characters is waiting to be displayed, Notion becomes unresponsive (you can't scroll).  
This occurs on pages greater than a character count of 2000. The larger a page gets, the worse the condition seems to get.

**expected behaviour**
I've tested out the same pages in the regular Notion desktop app and they exhibit no lag regardless of window size. I tested pages that are well beyond 2000 characters which means it should not be Notion's fault, as it seems to be able to handle heavier ones.

**screenshots**
Attached are gifs showing the lag. First is the enhanced version, the second is un-enhanced.
![Enhanced](https://user-images.githubusercontent.com/74338929/101968943-cdbefb00-3bef-11eb-99de-c2c5fe43581c.gif)
![Vanilla](https://user-images.githubusercontent.com/74338929/101968947-d0b9eb80-3bef-11eb-87cd-3874e3e4dcf1.gif)

**platform**
Windows 10, Notion Enhancer v0.10.2, Notion v2.0.11

**Comment by kursorbyte** [comment]
Update: The more you zoom out with **ctrl -**, and the smaller the text is, the lag becomes more reduced.

**Comment by Tkowns** [comment]
any solutions for this? I tried zooming out it still lags a little. would be awesome if typing can be smooth cuz the notion enhancer is awesome to use.

**Comment by rockmanvnx6** [comment]
@dragonwocky Please specify the other issue number for us to follow if this is marked as duplicated.

**Comment by dragonwocky** [comment]
@rockmanvnx6 v0.11.0 will feature a complete rewrite, which will have the side effect of fixing a lot of bugs along the way (and creating new ones). Since I have no way to know exactly what bugs will be fixed, particularly if I don't know what the exact cause of the bug is in the first place like this one, I'm cutting down on the number of issues here by generalising them to what they're likely related to or what I can actually fix, in this case https://github.com/notion-enhancer/notion-enhancer/issues/333. When v0.11.0 is released, all issues marked as a bug (excluding one or two that I am 100% certain will still exist) will be closed regardless, and new issues should be created if they recur.

TLDR: I am rewriting the enhancer, bugs are not likely to be carried over, I've closed all issues except ones I can actually focus on fixing.

**Comment by dragonwocky** [comment]
A sidenote: this is quite possibly caused by the theming system - even in the update I'm working on, there's a significantly noticeable increase in lag when theming is enabled. This is mainly because Notion renders everything stupidly, meaning theming requires a lot of CSS with some unusual specificity that makes the browser/app do a LOT of heavy lifting to override all Notion's styles. Modifying anything on page e.g. opening the sidebar or adding new content re-renders the page and forces the browser to re-do a fair bit of that work. This isn't particularly fixable, unfortunately.

**Comment by jarajto** [comment]
> Update: The more you zoom out with **ctrl -**, and the smaller the text is, the lag becomes more reduced.

Did you find out soultion or it is just enhancer?

**Comment by marwanm-dev** [comment]
@dragonwocky This problem is still here without themes.. Can you update us?

**Comment by dragonwocky** [comment]
@marwanm-dev there are many possible sources of lag. Some are unavoidable, some are bugs, some I gradually improve with each update. Unless a lag-related issue can be linked to a specific bug to be fixed, e.g. "this feature of this extension is causing X to happen," there's nothing that I can do about it other than what I'm already doing. Please don't ping closed issues.
